apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -565,7 +565,15 @@ public void recycle(TraceContext traceContext) {
         spanLinkPool.recycle(traceContext);
     }
 
-    public synchronized void stop() {
+    public void stop() {
+        stop(false);
+    }
+
+    public void stopForTest() {
+        stop(true);
+    }
+
+    private synchronized void stop(boolean fromTest) {
```

Comment:
```suggestion
    public synchronized void stop(boolean shutdownLogging) {
```

There is only a few cases where we actually shutdown the logging, so we'd rather make `stop(){...}` variant NOT stop logging as it's mostly used for tests.

Reply:
I did that originally because fewer changes like you say, but it looked clearer to explicitly say in the test that it's a test usage of stop. I'm not convinced my choice was better, but I don't think that just because there are fewer changes makes it better to do it the other way?

Reply:
lol, doesn't matter given the other suggestion